tcms/search/__init__.py

```diff
@@ -76,47 +73,56 @@ def _query(plan_query, run_query, case_query, target):
 
 
 def _sum_orm_queries(plans, cases, runs, target):
-    plans = plans.evaluate()
-    cases = cases.evaluate()
-    runs = runs.evaluate()
     if target == 'run':
-        if not plans and not cases:
-            if runs is None:
-                runs = TestRun.objects.none()
-        if runs is None:
-            runs = TestRun.objects.all()
-        if cases:
-            runs = runs.filter(case_run__case__in=cases).distinct()
-        if plans:
-            runs = runs.filter(plan__in=plans).distinct()
-        return runs
+        return _get_run_target(plans.evaluate(), cases.evaluate(), runs.evaluate())
```

Comment:
I think `.evaluate()` should stay as it was before so we don't repeat it everywhere. And then the calls to the 3 helper functions look fine.